ouroboros-consensus-test/src/Test/Util/Orphans/Isomorphism.hs

```diff
@@ -0,0 +1,125 @@
+{-# LANGUAGE FlexibleContexts        #-}
+{-# LANGUAGE FlexibleInstances       #-}
+{-# LANGUAGE InstanceSigs            #-}
+{-# LANGUAGE LambdaCase              #-}
+{-# LANGUAGE MultiParamTypeClasses   #-}
+{-# LANGUAGE UndecidableSuperClasses #-}
+
+module Test.Util.Orphans.Isomorphism (
+    Isomorphism (..)
+  , from
+  , inside
+  ) where
+
+import           Data.Foldable (toList)
+import qualified Data.Sequence as Seq
+
+import qualified Data.FingerTree.Strict as FT
+
+import qualified Data.FingerTree.TopMeasured.Strict as TMFT
+import qualified Data.Map.Diff.Strict as MapDiff
+import qualified Ouroboros.Consensus.Block as Block
+import qualified Ouroboros.Consensus.Storage.LedgerDB.HD as HD
+import qualified Ouroboros.Consensus.Storage.LedgerDB.HD.DiffSeq as DS
+import qualified Ouroboros.Consensus.Storage.LedgerDB.HD.TableTypes as TT
+
+{------------------------------------------------------------------------------
+  @'Isomorphism'@ class
+------------------------------------------------------------------------------}
+
+-- FIXME(jdral): This class is based on the @IsomorphicTo@ class from REF. We
+-- can not use the package at this time because the cabal update index is too
+-- outdated. Once the index has been updated to a more recent one, we should
+-- switch to this package.
+--
+-- REF: https://hackage.haskell.org/package/isomorphism-class-0.1.0.5/docs/IsomorphismClass.html
```

Comment:
Should be fixable when `feature/utxo-hd` is rebased on the most recent version of `master`.